contracts/contracts/MessageProcessor.sol

```diff
@@ -67,6 +70,8 @@ contract MessageProcessor is Ownable, SnarkCommon, Hasher, CommonUtilities, IMes
     vkRegistry = IVkRegistry(_vkRegistry);
     poll = IPoll(_poll);
     mode = _mode;
+    batchHashes = poll.getBatchHashes();
```

Comment:
I think here we can save some gas and hardcode it to the value we expect from the poll init message

Reply:
resolved